### ISSUE TITLE
Create architecture Wiki page

### ISSUE THREAD

**Issue by kdhageman** [aau-network-security/haaukins#227]

**Comment by kidmose** [comment]
If you need feedback on this, I can do that. (And if need be, I can also contribute more)

**Comment by mrturkmen06** [comment]
Just making some sketches on architecture of Haaukins using draw.io, so some feedback based on that could be very useful.

**Comment by mrturkmen06** [comment]
Initially, the components of Haaukins represented as shown diagram below, any feedback would be very useful for me to make further improvements...
 
<img width="1534" alt="arch" src="https://user-images.githubusercontent.com/13614433/63258464-618eea00-c27c-11e9-8706-6ee40e60b40d.png">

**Comment by kidmose** [comment]
Ah. I assumed you were working on the planned architecture, as discussed on our meeting last week ([link](https://share.icloud.com/photos/0o0oPpIdE84nMSncTT3TKWt2A))
Capturing the as-is seems like a good thing to do before that :)
But in that case, I'm not as capable of providing useful feedback.

Minor, spellcheck: Exercies -> Exercise
Generally: Makes sense :+1:

**Comment by mrturkmen06** [comment]
Ups, I thought that enrichment of wiki page with diagrams and explanations, based on current status of Haaukins, will take place with this issue. Since there is no "new" architecture statement anywhere :) I might misunderstand the issue, if so, correction will definitely help me to find out correct way @kidmose. 
BTW, we will discuss on this with Kaspar, ASAP.

**Comment by mrturkmen06** [comment]
In order to give easy understanding of Haaukins' architecture, there will be some diagrams, each of them will focus on different aspects on Haaukins. Primarily, the mentioned diagrams will be created;

- Communication Flow within Haaukins 
- Connection model between components (such as 1-to-1 , 1-to-2, 1-to-many, many-to-many)
- Detailed redirects between Proxy,  CTFd and Apache Guacamole  module  

In addition to described diagrams, any other suggestions or making correction on described diagrams are most welcome ... :) It would be very beneficial for us to give feedback, which will lead upcoming developments and improvements on Haaukins.

**Comment by mrturkmen06** [comment]
Communication flow within Haaukins illustrated with following figure below 

<img width="1018" alt="Communication Flow" src="https://user-images.githubusercontent.com/13614433/63433093-c8e69e80-c422-11e9-9584-6e43219dbfef.png">

Advice or any comment based on that would be beneficial :)

**Comment by mrturkmen06** [comment]
I faced some difficulties to define subtitles for architecture wiki page, however I have tried to write down some explanation with additional diagrams into word page, I can call it as draft version of actual architecture wiki page, so addition, comment and correction is always well welcome. I have created "Drafted In-Progress " column under [here](https://github.com/orgs/aau-network-security/projects/1) and moved this issue into drafted-in-progress column.  The name of column is a little bit weird however I think it is good way to go. So, according to needs and comments wiki page will be updated during process, draft version of it is accessible from this [link](https://1drv.ms/w/s!AufrJhaLfGmqhJFIBlWttKtBUF07Ww)
